File: Others/ZOJ/2736_Daffodil_number.py
while True:
    try:
        n = input()
    except EOFError:
        break
    a, b, c = map(int, list(n))
    if int(n) == a ** 3 + b ** 3 + c ** 3:
        print('Yes')
    else:
        print('No')

File: Others/ZOJ/2736_Daffodil_number.cpp
#include <bits/stdc++.h>
using namespace std;

int main(){
    int n, a, b, c;
    while (cin >> n){
        a = n / 100;
        b = n / 10 % 10;
        c = n % 10;
        if (n == pow(a, 3) + pow(b, 3) + pow(c, 3)) {
            cout << "Yes" << endl;
        } else {
            cout << "No" << endl;
        }
    }
    return 0;
}